### PR TITLE
Add entry points

### PR DESCRIPTION
**Pull request by gsheni**
- Added an entrypoint for featuretools primitives

### CODE REVIEW

File: featuretools/primitives/__init__.py [hunk 1]
```diff
@@ -1,2 +1,8 @@
 # flake8: noqa
 from .api import *
+
+import pkg_resources
+for entry_point in pkg_resources.iter_entry_points('featuretools_primitives'):
+    if hasattr(entry_point, 'primitives'):
```
Comment (kmax12) [inline featuretools/primitives/__init__.py]: should this be `hasattr(entry_point.load(), 'primitives')`?

Comment (gsheni) [reply featuretools/primitives/__init__.py]: Fixed.

File: featuretools/primitives/__init__.py [hunk 2]
```diff
@@ -1,2 +1,12 @@
 # flake8: noqa
 from .api import *
+
+import pkg_resources
+for entry_point in pkg_resources.iter_entry_points('featuretools_primitives'):
```
Comment (kmax12) [inline featuretools/primitives/__init__.py]: add a comment to this block? something like 

"Load in primitives registered by other libraries into Featuretools namespace"

File: featuretools/synthesis/dfs.py [hunk 3]
```diff
@@ -174,6 +175,15 @@ def dfs(entities=None,
     if not isinstance(entityset, EntitySet):
         entityset = EntitySet("dfs", entities, relationships)
 
+    for entry_point in pkg_resources.iter_entry_points('featuretools_dfs'):
```
Comment (kmax12) [inline featuretools/synthesis/dfs.py]: add a comment to this block? something like

"Call functions registered by other libraries with DFS arguments"

File: featuretools/synthesis/dfs.py [hunk 4]
```diff
@@ -174,6 +175,15 @@ def dfs(entities=None,
...
+        try:
+            loaded = entry_point.load()
+            # TODO: change name of module function
```
Comment (kmax12) [inline featuretools/synthesis/dfs.py]: this module function works for now

File: featuretools/tests/entry_point_tests/test_entry_point.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
+# import pandas as pd
```
Comment (kmax12) [inline featuretools/tests/entry_point_tests/test_entry_point.py]: delete this file?

File: featuretools/__init__.py [hunk 6]
```diff
@@ -16,3 +16,20 @@
...
+for entry_point in pkg_resources.iter_entry_points('featuretools_initialize'):
+    try:
+        module = entry_point.load()
+        if hasattr(module, 'initialize'):
+            module.initialize()
+    except Exception:
+        pass
+
+
+for entry_point in pkg_resources.iter_entry_points('featuretools_plugin'):
```
Comment (kmax12) [inline featuretools/__init__.py]: add a comment to this block? something like

"Load in submodules registered by other libraries into Featuretools namespace"

File: featuretools/__init__.py [hunk 7]
```diff
@@ -16,3 +16,20 @@
 from .feature_base import AggregationFeature, DirectFeature, Feature, FeatureBase, IdentityFeature, TransformFeature
 
 __version__ = '0.6.1'
+
+import pkg_resources
+import sys
+for entry_point in pkg_resources.iter_entry_points('featuretools_initialize'):
```
Comment (kmax12) [inline featuretools/__init__.py]: add a comment to this block? something like

"Call functions registered by other libraries when featuretools is imported"

File: featuretools/entityset/serialize.py [hunk 8]
```diff
@@ -1,3 +1,4 @@
+
```
[the remark this answers is not in the document]
Comment (gsheni) [reply featuretools/entityset/serialize.py]: Fixed

File: featuretools/synthesis/__init__.py [hunk 9]
```diff
@@ -1,3 +1,4 @@
 from __future__ import absolute_import
 # flake8: noqa
 from .api import *
+
```
[the remark this answers is not in the document]
Comment (gsheni) [reply featuretools/synthesis/__init__.py]: Fixed.

File: featuretools/primitives/__init__.py (hunk 2 above)
Comment (gsheni) [reply featuretools/primitives/__init__.py]: Added

File: featuretools/synthesis/dfs.py (hunk 3 above)
Comment (gsheni) [reply featuretools/synthesis/dfs.py]: Added

File: featuretools/synthesis/dfs.py (hunk 4 above)
Comment (gsheni) [reply featuretools/synthesis/dfs.py]: Removed

File: featuretools/tests/entry_point_tests/test_entry_point.py (hunk 5 above)
Comment (gsheni) [reply featuretools/tests/entry_point_tests/test_entry_point.py]: Removed

File: featuretools/__init__.py (hunk 6 above)
Comment (gsheni) [reply featuretools/__init__.py]: Added

File: featuretools/__init__.py (hunk 7 above)
Comment (gsheni) [reply featuretools/__init__.py]: Added

**Pull request merged**